### PR TITLE
Use any v5 version of actions/setup-python

### PR DESCRIPTION
**Pull request by peterbe**

### CODE REVIEW

**Comment by peterbe** [comment]
@hartwork ?

**Comment by hartwork** [comment]
> @hartwork ?

@peterbe please see https://github.com/peterbe/hashin/pull/201#pullrequestreview-2399170402

**Review by hartwork — APPROVED** [review]
@peterbe Thanks for splitting #202 off of this!

I'm approving now based on…
- your conscious decision to fully trust future changes to `actions/checkout` over time
- my hope that you will either squash the three commits before merge or use squash merge for this pull request :pray:

**Pull request merged**
